nix/all-toplevel-derivations.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, pkgSet
+, fn
+, recursionDepth
+, keyFilter
+, ...
+}:
+let
```

Comment:
I think a comment about what this is supposed to do and a rough description of the algorithm would be good.

Reply:
done